src/render_helpers/surface.rs

```diff
@@ -53,17 +55,37 @@ pub fn render_snapshot_from_surface_tree(
                 }
 
                 let data = data.lock().unwrap();
-                let Some(texture) = data.texture(renderer.context_id()) else {
-                    return;
-                };
+                let buffer = {
+                    if let Some(texture) = data.texture::<GlesTexture>(renderer.context_id()) {
+                        TextureBuffer::from_texture(
+                            renderer,
+                            texture.clone(),
+                            f64::from(data.buffer_scale()),
+                            data.buffer_transform(),
+                            Vec::new(),
+                        )
+                    } else if let Some(single_pixel_buffer_user_data) = data
+                        .buffer()
+                        .and_then(|buffer| get_single_pixel_buffer(buffer).ok())
+                    {
+                        let pixel: [u8; 4] = single_pixel_buffer_user_data.rgba8888();
+                        let argb_pixel = [pixel[3], pixel[1], pixel[2], pixel[0]];
 
-                let buffer = TextureBuffer::from_texture(
-                    renderer,
-                    texture.clone(),
-                    f64::from(data.buffer_scale()),
-                    data.buffer_transform(),
-                    Vec::new(),
-                );
+                        TextureBuffer::from_memory(
+                            renderer,
+                            &argb_pixel,
+                            Fourcc::Argb8888,
```

Comment:
Maybe we can use Abgr and skip the pixel swapping?

Reply:
Do you mean that we should use `Rgba8888`? I tried doing that initially but it turns out that the supported formats for `GlesRenderer` are

```rust
const SUPPORTED_MEM_FORMATS_2: &[Fourcc] = &[
    Fourcc::Abgr8888,
    Fourcc::Xbgr8888,
    Fourcc::Argb8888,
    Fourcc::Xrgb8888,
];

const SUPPORTED_MEM_FORMATS_3: &[Fourcc] = &[
    Fourcc::Abgr8888,
    Fourcc::Xbgr8888,
    Fourcc::Argb8888,
    Fourcc::Xrgb8888,
    Fourcc::Abgr2101010,
    Fourcc::Xbgr2101010,
    Fourcc::Abgr16161616f,
    Fourcc::Xbgr16161616f,
];
```

If you mean use `Abgr8888` and just `.reverse` the pixel from the `SinglePixelBufferUserData` then that also sounds like a good idea.

Either way I'll definitely add a comment here